src/OSPSuite.Presentation/Presenters/Charts/ChartEditorPresenter.cs

```diff
@@ -278,20 +283,20 @@ private void assignSameColorToGroupedCurves(List<IReadOnlyList<DataRepository>>
       {
          foreach (var group in groupedDataRepositories)
          {
-            var groupColor = Chart.SelectNewColor();
             var curvesInGroup = findCurvesCorrespondingToRepositoryList(group);
 
             if (!curvesInGroup.Any())
                continue;
-            
+
+            var groupColor = Chart.SelectNewColor();
```

Comment:
I think the color should be used only if there are curves. Otherwise you will skip a color

Reply:
agree